### PR TITLE
Swagger integration 

### PR DESCRIPTION
**Pull request by Rupank**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -38,15 +53,20 @@
     "express": "^4.17.1",
     "figlet": "^1.2.4",
     "helmet": "^3.21.2",
+    "jest": "^24.9.0",
     "jsonfile": "^5.0.0",
     "microframework": "^0.6.4",
+    "microframework-w3tec": "^0.6.3",
```
Comment (VivekFitkariwala) [inline package.json]: Remove this dependency

File: src/loaders/swaggerLoader.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import { MicroframeworkLoader, MicroframeworkSettings } from 'microframework';
+import * as swaggerUi from 'swagger-ui-express';
+// tslint:disable-next-line:no-var-requires
+const swaggerJSDoc = require('swagger-jsdoc');
```
Comment (VivekFitkariwala) [inline src/loaders/swaggerLoader.ts]: Change this to import

File: src/loaders/swaggerLoader.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+                info: {
+                    title: env.app.name,
+                    description: env.app.description,
+                    servers: [{
+                        url: `${env.app.schema}://${env.app.host}:${env.app.port}${env.app.routePrefix}`,
+                    }],
+                },
+            },
+            explorer: true,
+            apis: ['**/*.ts'],
```
Comment (VivekFitkariwala) [inline src/loaders/swaggerLoader.ts]: Add only controllers over here

File: src/env.ts [hunk 4]
```diff
@@ -53,6 +53,12 @@ export const env = {
     synchronize: toBool(getOsEnvOptional('TYPEORM_SYNCHRONIZE')),
     logging: getOsEnv('TYPEORM_LOGGING'),
   },
-  swagger: {},
+  swagger: {
+    enabled: toBool(getOsEnv('SWAGGER_ENABLED')),
+    route: getOsEnv('SWAGGER_ROUTE'),
+    file: getOsEnv('SWAGGER_FILE'),
+    username: getOsEnv('SWAGGER_USERNAME'),
+    password: getOsEnv('SWAGGER_PASSWORD'),
```
Comment (VivekFitkariwala) [inline src/env.ts]: Are we using this at present?

File: test/utils/database.ts [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+export const createDatabaseConnection = async (): Promise<Connection> => {
+    useContainer(Container);
+    const connection = await createConnection({
+        type: env.db.type as any, // See createConnection options for valid types
+        database: env.db.database,
+        logging: env.db.logging as LoggerOptions,
+        entities: env.app.dirs.entities,
+        migrations: env.app.dirs.migrations,
+    });
+    console.log('Rupank', connection);
```
Comment (VivekFitkariwala) [inline test/utils/database.ts]: Use Logger class

File: package.json (hunk 1 above)
Comment (Rupank) [reply package.json]: @VivekFitkariwala Removed the unnecessary dependency from the code.

File: src/loaders/swaggerLoader.ts (hunk 2 above)
Comment (Rupank) [reply src/loaders/swaggerLoader.ts]: @VivekFitkariwala Changed to import statement.

File: src/loaders/swaggerLoader.ts (hunk 3 above)
Comment (Rupank) [reply src/loaders/swaggerLoader.ts]: @VivekFitkariwala Added only controllers path in the apis array.

File: test/utils/database.ts (hunk 5 above)
Comment (Rupank) [reply test/utils/database.ts]: @VivekFitkariwala Removed the log statement as it is not required at the moment.

**Pull request merged**